scripts/us_census/pep/monthly_population/download.py

```diff
@@ -0,0 +1,153 @@
+""" A Script to download USA Census PEP monthly population data
+    from the URLS in provided json file.
+    Typical usage:
+    1. python3 download.py
+    2. python3 download.py -f file_urls.json
+    3. python3 download.py -f <<path_to_file>>
+"""
+import os
+import json
+import argparse
+import pandas as pd
+import numpy as np
+
+HEADER = 1
+SKIP_ROWS = 1
+
+
+def save_data(url, download_local_path):
+    """
+    This method loads the Data from url to pandas Dataframe.
+    Writes the data to local path provided as one the parameter.
+
+    Parameters:
+    url: Url of the dataset
+    download_local_path: Path to save the datasets.
+
+    Returns:
+    None
+    """
+    df = ""
+    file_name = url.split("/")[-1]
+    if ".xls" in url:
+        df = pd.read_excel(url, header=HEADER)
+        df.to_excel(download_local_path + os.sep + file_name,
+                    index=False,
+                    header=False,
+                    engine='xlsxwriter')
+    elif ".txt" in url:
+        file_name = file_name.replace(".txt", ".xlsx")
+        cols = [
+            "Year and Month", "Date", "Resident Population",
+            "Resident Population Plus Armed Forces Overseas",
+            "Civilian Population", "Civilian NonInstitutionalized Population"
+        ]
+        df = pd.read_table(url,
+                           index_col=False,
+                           delim_whitespace=True,
+                           engine='python',
+                           skiprows=17,
+                           names=cols)
+        df = clean_txt_file(df)
+        df.to_excel(download_local_path + os.sep + file_name,
+                    index=False,
+                    engine='xlsxwriter')
+
+
+def sum_cols(col):
+    """
+    This method adds Dataframe columns Year and Month, Date.
+    Returns the column value in format yyyymmdd. Example: 2022March01
+    """
+    if col[1] is None:

```

Comment:
Please describe the what column indices are expected and what they contain.

If such functions are only used within this script and not by other scritps outside, the function name can be prefixed with `_`, say `def _sum_cols(col)`

Reply:
Ok, We will prefix with _ for such methods.